base/ca/src/main/java/org/dogtagpki/server/ca/cli/CACertRequestCLI.java

```diff
@@ -17,4 +19,21 @@ public CACertRequestCLI(CLI parent) {
 
         addModule(new CACertRequestImportCLI(this));
     }
+
+    public static void printCertRequestInfo(CertRequestInfo info) {
+
+        System.out.println("  Request ID: " + info.getRequestID().toHexString());
+        System.out.println("  Type: " + info.getRequestType());
+        System.out.println("  Status: " + info.getRequestStatus());
+        System.out.println("  Result: " + info.getOperationResult());
+
+        String error = info.getErrorMessage();
+        if (error != null) {
+            System.out.println("  Reason: " + error);
+        }
+
+        if (info.getCertId() != null) {
+            System.out.println("  Certificate ID: " + info.getCertId().toHexString());
+        }
```

Comment:
Alternatively, you could use `StringBuilder` and add these two lines conditionally using `Optional` which would save the additional if statements. The downside of that is `StringBuilder` can make difficult to read, Shakespearean-verbose code - it would certainly be uglier that using a text block!

I think I have spent enough time thinking about strings for one morning :laughing:

Reply:
I need to read about `Optional` too.